internal/interaction/repository_moq_test.go

```diff
@@ -289,6 +303,46 @@ func (mock *RepositoryMock) DeleteTransactionCalls() []struct {
 	return calls
 }
 
+// GetAdminTransactionsByFilter calls GetAdminTransactionsByFilterFunc.
+func (mock *RepositoryMock) GetAdminTransactionsByFilter(ctx context.Context, query entities.TransactionQuery) ([]entities.Transaction, error) {
+	callInfo := struct {
+		Ctx   context.Context
+		Query entities.TransactionQuery
+	}{
+		Ctx:   ctx,
+		Query: query,
+	}
+	mock.lockGetAdminTransactionsByFilter.Lock()
+	mock.calls.GetAdminTransactionsByFilter = append(mock.calls.GetAdminTransactionsByFilter, callInfo)
+	mock.lockGetAdminTransactionsByFilter.Unlock()
```

Comment:
afaik the Unlock() call should always be done with defer so it will happen even in the event of a panic.

Reply:
This is a generated file and should not be edited

Reply:
ah ok.